ansible-lint build fails w/ PyYAML deprecation warning

**Describe the bug**
A deprecation warning fails the ansible-lint build.

**To Reproduce**
Steps to reproduce the behavior:
1. Install ansible-lint, python37Packages.ansible-lint, or python38Packages.ansible-lint

**Expected behavior**
Successful build

**Screenshots**
```
Installing collected packages: ansible-lint
Successfully installed ansible-lint-4.3.7
/build/ansible-lint-4.3.7
Finished executing pipInstallPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/2x2yh0c63z1y74w0dgnfw4af9zqg6dzm-python3.7-ansible-lint-4.3.7
strip is /nix/store/9f8y44vmjnwdjvzlff0gm3f3g6ycyyqy-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/2x2yh0c63z1y74w0dgnfw4af9zqg6dzm-python3.7-ansible-lint-4.3.7/lib  /nix/store/2x2yh0c63z1y74w0dgnfw4af9zqg6dzm-python3.7-ansible-lint-4.3.7/bin
patching script interpreter paths in /nix/store/2x2yh0c63z1y74w0dgnfw4af9zqg6dzm-python3.7-ansible-lint-4.3.7
checking for references to /build/ in /nix/store/2x2yh0c63z1y74w0dgnfw4af9zqg6dzm-python3.7-ansible-lint-4.3.7...
Rewriting #!/nix/store/yd4nslxqldgl2m2h53g3ig78far7vq4y-python3-3.7.9/bin/python3.7 to #!/nix/store/yd4nslxqldgl2m2h53g3ig78far7vq4y-python3-3.7.9
wrapping `/nix/store/2x2yh0c63z1y74w0dgnfw4af9zqg6dzm-python3.7-ansible-lint-4.3.7/bin/ansible-lint'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
ImportError while loading conftest '/build/ansible-lint-4.3.7/conftest.py'.
conftest.py:6: in <module>
    from ansiblelint.constants import DEFAULT_RULESDIR
/nix/store/2x2yh0c63z1y74w0dgnfw4af9zqg6dzm-python3.7-ansible-lint-4.3.7/lib/python3.7/site-packages/ansiblelint/__init__.py:22: in <module>
    from ansiblelint.rules import AnsibleLintRule
/nix/store/2x2yh0c63z1y74w0dgnfw4af9zqg6dzm-python3.7-ansible-lint-4.3.7/lib/python3.7/site-packages/ansiblelint/rules/__init__.py:12: in <module>
    import ansiblelint.utils
/nix/store/2x2yh0c63z1y74w0dgnfw4af9zqg6dzm-python3.7-ansible-lint-4.3.7/lib/python3.7/site-packages/ansiblelint/utils.py:37: in <module>
    from ansible.parsing.dataloader import DataLoader
/nix/store/kl4y2li380h6sahmj107mf2qbiwpjxpj-python3.7-ansible-2.9.12/lib/python3.7/site-packages/ansible/parsing/dataloader.py:21: in <module>
    from ansible.parsing.utils.yaml import from_yaml
/nix/store/kl4y2li380h6sahmj107mf2qbiwpjxpj-python3.7-ansible-2.9.12/lib/python3.7/site-packages/ansible/parsing/utils/yaml.py:17: in <module>
    from ansible.parsing.yaml.loader import AnsibleLoader
/nix/store/kl4y2li380h6sahmj107mf2qbiwpjxpj-python3.7-ansible-2.9.12/lib/python3.7/site-packages/ansible/parsing/yaml/loader.py:23: in <module>
    from _yaml import CParser, CEmitter
/nix/store/g44n0jp7mfd2vd3i7vp1z1fspqgdl63a-python3.7-PyYAML-5.4.1/lib/python3.7/site-packages/_yaml/__init__.py:23: in <module>
    DeprecationWarning
E   DeprecationWarning: The _yaml extension module is now located at yaml._yaml and its location is subject to change.  To use the LibYAML-based parser and emitter, import from `yaml`: `from yaml import CLoader as Loader, CDumper as Dumper`.
```

**Notify maintainers**
@sengaya 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.104-microsoft-standard, Ubuntu, 20.04.2 LTS (Focal Fossa)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(james): `"nixpkgs-21.05pre271444.9816b99e71c"`
 - nixpkgs: `/home/james/.nix-defexpr/channels/nixpkgs`
 - 
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - ansible-lint
  - python37Packages.ansible-lint
  - python38Packages.ansible-lint
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/python-modules/ansible-lint/default.nix
+++ b/pkgs/development/python-modules/ansible-lint/default.nix
@@ -43,7 +43,7 @@ buildPythonPackage rec {
   '';
 
   checkPhase = ''
-    pytest -k 'not test_run_playbook_github and not test_run_single_role_path_no_trailing_slash_script'
+    pytest -W ignore::DeprecationWarning -k 'not test_run_playbook_github and not test_run_single_role_path_no_trailing_slash_script'
   '';
 
   meta = with lib; {

